packages/core/src/external/elation/index.ts

```diff
@@ -0,0 +1,370 @@
+import { capture } from "../../util/notifications";
+import { uuidv7 } from "../../util/uuid-v7";
+import { S3Utils } from "../aws/s3";
+
+interface ApiConfig {
+  practiceId: string;
+  environment: ElationEnv;
+  clientKey: string;
+  clientSecret: string;
+}
+
+const region = Config.getAWSRegion();
+const responsesBucket = Config.getEhrResponsesBucketName();
+const elationDateFormat = "YYYY-MM-DD";
+
+function getS3UtilsInstance(): S3Utils {
+  return new S3Utils(region);
+}
+
+export type ElationEnv = "api" | "sandbox";
+
+class ElationApi {
+  private axiosInstanceProprietary: AxiosInstance;
+  private baseUrl: string;
+  private twoLeggedAuthToken: string;
+  private practiceId: string;
+  private s3Utils: S3Utils;
+
+  private constructor(private config: ApiConfig) {
+    this.twoLeggedAuthToken = "";
+    this.practiceId = config.practiceId;
+    this.s3Utils = getS3UtilsInstance();
+    this.axiosInstanceProprietary = axios.create({});
+    this.baseUrl = `https://${config.environment}.elationemr.com/api/2.0`;
+  }
+
+  public static async create(config: ApiConfig): Promise<ElationApi> {
+    const instance = new ElationApi(config);
+    await instance.initialize();
+    return instance;
+  }
+
+  private async fetchtwoLeggedAuthToken(): Promise<void> {
+    const url = `${this.baseUrl}/oauth2/token/`;
+    const data = {
+      grant_type: "client_credentials",
+      client_id: this.config.clientKey,
+      client_secret: this.config.clientSecret,
+    };
+
+    try {
+      const response = await axios.post(url, this.createDataParams(data), {
+        headers: { "content-type": "application/x-www-form-urlencoded" },
+      });
+
+      this.twoLeggedAuthToken = response.data.access_token;
+    } catch (error) {
+      throw new MetriportError("Failed to fetch Two Legged Auth token");
+    }
+  }
+
+  async initialize(): Promise<void> {
+    await this.fetchtwoLeggedAuthToken();
+
+    this.axiosInstanceProprietary = axios.create({
+      baseURL: this.baseUrl,
+      headers: {
+        Authorization: `Bearer ${this.twoLeggedAuthToken}`,
+        "content-type": "application/x-www-form-urlencoded",
+      },
+    });
+  }
+
+  async getPatient({
+    cxId,
+    patientId,
+  }: {
+    cxId: string;
+    patientId: string;
+  }): Promise<PatientResource | undefined> {
+    const { log, debug } = out(
+      `Elation get patient - cxId ${cxId} practiceId ${this.practiceId} patientId ${patientId}`
+    );
+    const patientUrl = `/patients/${patientId}/`;
+    try {
+      const response = await this.axiosInstanceProprietary.get(patientUrl);
+      if (!response.data) throw new MetriportError(`No body returned from ${patientUrl}`);
+      debug(`${patientUrl} resp: ${JSON.stringify(response.data)}`);
+      if (responsesBucket) {
+        const filePath = createHivePartitionFilePath({
+          cxId,
+          patientId,
+          date: new Date(),
+        });
+        const key = `elation/patient/${filePath}/${uuidv7()}.json`;
```

Comment:
will look.

Reply:
Not clear option here -- best I can do it a `${method}_${baseUrl}` type thing.

Reply:
No worries. We're just using UUID because we might have more than one request per second, otherwise we could just have a fixed name instead of the UUID based one, right?

All good, I just wanted to try and give some meaning to it, instead of a random UUID - I'm sure someone at some point will be like: "what does that ID represent?" 😄